var/spack/repos/builtin/packages/py-pycuda/package.py

```diff
@@ -24,6 +24,26 @@ class PyPycuda(PythonPackage):
 
     depends_on("cxx", type="build")  # generated
 
+    # TODO: replace this with an explicit list of components of Boost,
+    # for instance depends_on('boost +filesystem')
+    # See https://github.com/spack/spack/pull/22303 for reference
+    depends_on(Boost.with_default_variants)
+    depends_on("python@3.6:3", type=("build", "run"), when="@2020.1:")
+    depends_on("python@3.8:3", type=("build", "run"), when="@2024.1.2:")
+    depends_on("boost+python")
+    depends_on("cuda", type=("build", "run"))
+    depends_on("cuda@:8.0.61", when="@2016.1.2")
+    depends_on("py-appdirs@1.4.0:", type=("build", "run"))
+    depends_on("py-decorator@3.2.0:", type=("build", "run"), when="@:2020.1")
+    depends_on("py-mako", type=("build", "run"))
+    depends_on("py-numpy@1.6:", type=("build", "run"))
+    depends_on("py-numpy@1.24:", type=("build", "run"), when="@2024.1.2:")
```

Comment:
This should reflect the software versions where the `numpy` version constraint actually changed but, assuming the new minimum started with `2024.1`:

```suggestion
    depends_on("py-numpy@1.6:", type=("build", "run"), when="@:2024.1")
    depends_on("py-numpy@1.24:", type=("build", "run"), when="@2024.1:")
```